lib/collection/src/collection_builder/collection_builder_base.rs

```diff
@@ -22,13 +22,14 @@ pub fn construct_collection(
     segment_holder: SegmentHolder,
     config: CollectionConfig,
     wal: SerdeWal<CollectionUpdateOperations>,
-    optimizers: Arc<Vec<Box<Optimizer>>>,
+    optimizers: Arc<Vec<Arc<Optimizer>>>,
     collection_path: &Path,
 ) -> Collection {
     let segment_holder = Arc::new(RwLock::new(segment_holder));
 
     let optimize_runtime = runtime::Builder::new_multi_thread()
         .worker_threads(2)
+        .max_blocking_threads(2)
```

Comment:
we should probably make a configuration param for max threads. In might be a part of `CollectionConfig` and I guess the default value could be `num_cpu - 1`